### ISSUE TITLE
Projection stream position does not move from -1

### ISSUE THREAD

**Issue by StevenBlair123** [EventStore/EventStore#2954]
**Describe the bug**
When creating a projection which reads from a truncated stream ($et), the position on any of streams remains at -1

**To Reproduce**

1. Truncate a stream (tried using a Delete from UI and maxAge) - note the stream tends to have millions of truncated events.
2. Create a projection which uses this truncated stream
3. Projection will remain at Position -1

You can reset the system projections to get the same effect.

**Expected behavior**
Should be able to read from the streams.

**Actual behavior**
Projection does not move position.

**Config/Logs/Screenshots**
If applicable, please attach your node configuration, logs or any screenshots.

**Additional context**
https://discuss.eventstore.com/t/projection-stream-position-does-not-move-from-1/3232
https://github.com/EventStore/EventStore/issues/1872

**Comment by StevenBlair123** [comment]
It's been a while since this was originally spotted, but we think we have found the source of the problem.

Inside `StorageReaderWorker.ReadStreamEventsForward`, the code reads from the stream using `ReadStreamEventsForward`
The problem is when the stream has been truncated, the ReadStreamEventsForward fromEventNumber starts at 0.
When result is returned, the metadata indicates that the stream has indeed been truncated (TruncateBefore) and that seems to be the end of us trying to read events from here.

As a proof of concept, we have have added the following code in after the ReadStreamEventsForward:

```
if (result.Metadata != null && result.Metadata.TruncateBefore != null && result.Records.Length == 0) {
                        result =
                            _readIndex.ReadStreamEventsForward(streamName, streamId, result.NextEventNumber, msg.MaxCount);
                    }
```

So in short, if we have TruncatedBefore and no events, try again from result.NextEventNumber.

After applying this, our projections burst into life!

Any problems with this approach (we are going to clean the code up and create a PR shortly)?
